tests/Games/NexusMods.Games.TestFramework/AModInstallerTest.cs

```diff
@@ -70,13 +76,32 @@ protected ulong NextHash()
         AbsolutePath archivePath,
         CancellationToken cancellationToken = default)
     {
-        var analyzedArchive = await AnalyzeArchive(archivePath);
+        var downloadId = await DownloadRegistry.RegisterDownload(archivePath, new FilePathMetadata
+        {
+            OriginalName = archivePath.FileName,
+            Quality = Quality.Low
+        }, cancellationToken);
+
+        var contents = await DownloadRegistry.Get(downloadId);
+
+        var tree = FileTreeNode<RelativePath, ModSourceFileEntry>.CreateTree(contents.Contents
+            .Select(f =>
+                KeyValuePair.Create(f.Path,
+                    new ModSourceFileEntry
+                    {
+                        Hash = f.Hash,
+                        Size = f.Size,
+                        StreamFactory = new ArchiveManagerStreamFactory(ArchiveManager, f.Hash)
+                        {
+                            Name = f.Path,
+                            Size = f.Size
+                        }
```

Comment:
Should this be made into an extension method `ToModSourceFileEntry(this IDownloadContentEntry)` ?
I have seen this code repeated a few times.

Reply:
I think half of the uses are in tests, so I'll leave them for now but we can add extensions later if we want.